eloop.c

```diff
@@ -522,16 +481,11 @@ void eloop_run(void)
 
 		if (res <= 0)
 			continue;
-
-		eloop_sock_table_dispatch(&eloop.readers, rfds);
-		eloop_sock_table_dispatch(&eloop.writers, wfds);
-		eloop_sock_table_dispatch(&eloop.exceptions, efds);
+		eloop_sock_table_dispatch(&eloop.sock_table, fds, POLLIN);
```

Comment:
Hardcoding `POLLIN` here feels funny, but it is faithful to the original code where specific tables were passed in and keeps the same logic flow in place. I don't really have a recommendation so we can keep this as is.